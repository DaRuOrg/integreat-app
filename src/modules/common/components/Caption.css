@import url('../constants/dimens.css');

.caption {
  margin: 25px 0;
  text-align: center;
}

@media (--small-viewport) {
  .caption {
    margin: 10px 0;
  }
}
