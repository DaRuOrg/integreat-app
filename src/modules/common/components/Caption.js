import * as React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native-elements'
import { View } from 'react-native'

// fixme: Caption is connected to redux state because of H1 -> not a plain old component

class Caption extends React.Component {
  static propTypes = {
    title: PropTypes.string.isRequired
  }

  render () {
    return <React.Fragment>
      <Text>'sdf'</Text>
      <Text>'sdf'</Text>
      <Text>'sdf'</Text>
      <View><Text>'sdf'</Text></View>
    </React.Fragment>
  }
}

export default Caption
