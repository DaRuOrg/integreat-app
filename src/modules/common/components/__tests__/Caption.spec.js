import React from 'react'
import { render, shallow, mount } from 'enzyme'
import Caption from '../Caption'

describe('Caption', () => {
  it('should render', () => {
    expect(shallow(<Caption title={'Test Title'} />)).toMatchSnapshot()
  })
})
