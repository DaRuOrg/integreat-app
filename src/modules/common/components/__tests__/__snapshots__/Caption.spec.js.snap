// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption should render 1`] = `
<React.Fragment>
  <TextElement>
    'sdf'
  </TextElement>
  <TextElement>
    'sdf'
  </TextElement>
  <TextElement>
    'sdf'
  </TextElement>
  <Component>
    <TextElement>
      'sdf'
    </TextElement>
  </Component>
</React.Fragment>
`;
